test/fw/ptl/utils/pbs_snaputils.py

```diff
@@ -0,0 +1,1777 @@
+        :param core_dir: path to directory to store core information
+        :type core_dir: str
+
+        :return bool: True if this was a valid core file, otherwise False
+        """
+        if os.path.isfile(file_path) is False:
+            return False
+
+        file_header = magic.from_file(file_path)
+        if "core file" not in file_header:
+            return False
+
+        # Identify the program which created this core file
+        header_list = file_header.split()
+        if "from" not in header_list:
+            return False
+        exec_index = header_list.index("from") + 1
+        exec_name = header_list[
+            exec_index].replace("\'", "")
+        exec_name = exec_name.replace(",", "")
+
+        # Capture the stack trace from this core file
+        filename = os.path.basename(file_path)
+        core_dest = os.path.join(core_dir, filename)
+        self.__capture_trace_from_core(file_path, exec_name,
+                                       core_dest)
+
+        # Delete the core file itself
+        if os.path.isfile(file_path):
+            os.remove(file_path)
+
+        return True
+
+    def __anonymize_file(self, file_path):
+        """
+        Anonymize/obfuscate a file to remove sensitive information
+
+        :param file_path: path to the file to anonymize
+        :type str
+        """
+        if self.anonymize is False or self.anon_obj is None:
+            return
+
+        self.logger.debug("Anonymizing " + file_path)
+
+        file_name = os.path.basename(file_path)
+        if file_name == "sched_config":
+            self.anon_obj.anonymize_sched_config(self.scheduler)
+            self.scheduler.apply_config(path=file_path, validate=False)
+        elif file_name == "resource_group":
+            anon = self.anon_obj.anonymize_resource_group(file_path)
+            if anon is not None:
+                with open(file_path, "w") as rgfd:
+                    rgfd.write("\n".join(anon))
+        else:
+            self.anon_obj.anonymize_file(file_path, inplace=True)
+
+    def __copy_dir_with_core(self, host, src_path, dest_path, core_dir,
+                             except_list=[], only_core=False):
+        """
+        Copy over a directory recursively which might have core files
+        When a core file is found, capture the stack trace from it
+
+        :param host: name of the host where the source is located
+        :type str
+        :param src_path: path of the source directory
+        :type str
+        :param dest_path: path of the destination directory
+        :type str
+        :param core_dir: path to the directory to store core files' trace
+        :type str
+        :param except_list: list  of files/directories (basenames) to exclude
+        :type list
+        :param only_core: Copy over only core files?
+        :type bool
+        """
+        dir_list = None
+        try:
+            dir_list = self.du.listdir(host, src_path, fullpath=False,
```

Comment:
listdir could previously raise exceptions as it was directly calling os.listdir(), I've added a try-except inside there now.